x-pack/test/functional/apps/infra/hosts_view.ts

```diff
@@ -168,6 +168,16 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
         (await pageObjects.infraHostsView.isKPIChartsLoaded())
     );
 
+  const refreshPageWithDelay = async () => {
+    /**
+     * Delay gives React a chance to finish
+     * running effects (like updating the URL) before
+     * refreshing the page.
+     */
+    await pageObjects.common.sleep(1000);
```

Comment:
Q: Have you encountered any problems due to a delay in updating the state before refreshing the page?

Reply:
Yeah, unfortunately sometimes the browser starts to refresh before URL is changed. I assume it's because we don't write to the URL directly but rather use router hooks, making URL updates basically part of the react rendering cycle. Which is convenient for business logic of course, but for these kinds of tests I could not find a better solution other than to introduce an artificial delay, unfortunately 😔